core/trino-main/src/test/java/io/trino/sql/query/QueryAssertions.java

```diff
@@ -526,6 +527,21 @@ public QueryAssert isNotFullyPushedDown(PlanMatchPattern retainedSubplan)
             });
         }
 
+        /**
+         * Verifies join query is not fully pushed down by containing JOIN node.
+         */
+        public QueryAssert joinIsNotFullyPushedDown()
```

Comment:
I was able to rearrange commits in this way:
1) query assertion changes
2) new outer tests + `assertJoinConditionallyPushedDown` (cannot do assertJoinConditionallyPushedDown first because new tests use this assert immediately, without it it will not compile and no alternative to replace it in loop for all cases)
3)  `assertJoinConditionallyPushedDown` for all existing usages
4) disable OUTER join in postgres

Not sure maybe I should squash 2-3-4, because tests for postgres will not pass separately for each commit (but code will compile)

Reply:
Postgres should be squashed with 2 since the OUTER tests would fail without it. Rest of it looks good.

Reply:
Done